File: C#软件Dll加密混淆工具/SDK/IntelliLock Example Projects/Compact Framework/Licensed Application/C#/VS 2008/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Reflection;
using System.Windows.Forms;
using IntelliLock.Licensing;

namespace SDK_TestApp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
                        
            PropertyInfo[] properties = typeof(IntelliLock.Licensing.License).GetProperties();

            for (int i = 0; i < properties.Length; i++)
            {
                this.propertyComboBox.Items.Add(properties[i].Name);
            }

            if (properties.Length > 0)
            this.propertyComboBox.SelectedIndex = 0;

            hardwareidlabel.Text = IntelliLock.Licensing.HardwareID.GetHardwareID(true, true, true, true, true, true);

            this.listView1.Items.Clear();

            if ((IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseStatus == IntelliLock.Licensing.LicenseStatus.EvaluationMode) || (IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseStatus == IntelliLock.Licensing.LicenseStatus.Licensed))
            {
                for (int i = 0; i < IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseInformation.Count; i++)
                {
                    this.listView1.Items.Add(new ListViewItem(new string[] { IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseInformation.GetKey(i).ToString(), IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseInformation.GetByIndex(i).ToString() }));
                }
            }
        }

        private void propertyComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (propertyComboBox.SelectedIndex >= 0)
            {
                string propertyname = (string)propertyComboBox.Items[propertyComboBox.SelectedIndex];
                PropertyInfo propinfo = typeof(IntelliLock.Licensing.License).GetProperty(propertyname);
                valuelabel.Text = propinfo.GetValue(IntelliLock.Licensing.EvaluationMonitor.CurrentLicense, new object[0]).ToString();
            }
        }

        private void menuItem2_Click(object sender, EventArgs e)
        {
            MessageBox.Show("Deactivation Code:\r\n" + IntelliLock.Licensing.License_DeActivator.DeactivateLicense());
        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: C#软件Dll加密混淆工具/SDK/IntelliLock Example Projects/Compact Framework/Licensed Application/VB.NET/VS 2008/Form1.vb
Public Class Form1
    Inherits Form

    Public Sub New()
        Dim num As Integer
        Me.components = Nothing
        Me.InitializeComponent()

        Dim properties As System.Reflection.PropertyInfo() = GetType(IntelliLock.Licensing.License).GetProperties

        For num = 0 To properties.Length - 1
            Me.PropertyComboBox.Items.Add(properties(num).Name)
        Next num

        If (properties.Length > 0) Then
            Me.PropertyComboBox.SelectedIndex = 0
        End If

        Me.hardwareidlabel.Text = IntelliLock.Licensing.HardwareID.GetHardwareID(True, True, True, True, True, True)
        Me.ListView1.Items.Clear()

        If ((IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseStatus = IntelliLock.Licensing.LicenseStatus.EvaluationMode) OrElse (IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseStatus = IntelliLock.Licensing.LicenseStatus.Licensed)) Then
            For num = 0 To IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseInformation.Count - 1
                Me.ListView1.Items.Add(New ListViewItem(New String() {IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseInformation.GetKey(num).ToString, IntelliLock.Licensing.EvaluationMonitor.CurrentLicense.LicenseInformation.GetByIndex(num).ToString}))
            Next num
        End If

    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        Me.Close()
    End Sub

    Private Sub PropertyComboBox_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles PropertyComboBox.SelectedIndexChanged
        If (Me.PropertyComboBox.SelectedIndex >= 0) Then
            Dim name As String = CStr(Me.PropertyComboBox.Items.Item(Me.PropertyComboBox.SelectedIndex))
            Dim _property As System.Reflection.PropertyInfo = GetType(IntelliLock.Licensing.License).GetProperty(name)
            Me.valuelabel.Text = [_property].GetValue(IntelliLock.Licensing.EvaluationMonitor.CurrentLicense, New Object(0 - 1) {}).ToString
        End If
    End Sub

    Private Sub MenuItem2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)

    End Sub

    Private Sub MenuItem2_Click_1(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MenuItem2.Click
        MessageBox.Show(("Deactivation Code:" & ChrW(13) & ChrW(10) & IntelliLock.Licensing.License_DeActivator.DeactivateLicense))
    End Sub
End Class
